### ISSUE TITLE
[no-restricted-imports] No way to use this rule with `import = require()` syntax.

### ISSUE THREAD

**Issue by rtruong** [typescript-eslint/typescript-eslint#2563]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-restricted-imports": [
      "error",
      {
        paths: ["lodash"],
      },
    ],
  }
}
```

```TS
import _ = require("lodash");
```

**Expected Result**

Should result in a linting error.

**Actual Result**

No errors.

**Additional Info**

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `4.1.0` |
| `@typescript-eslint/parser`             | `4.1.0` |
| `@typescript-eslint/typescript-estree`  | `4.1.0` |
| `@typescript-eslint/experimental-utils` | `4.1.0` |
| `TypeScript`                            | `3.5.2` |
| `node`                                  | `12.16.3` |
| `npm`                                   | `6.14.7` |

**Comment by bradzacher** [comment]
See related request: #2403

**Comment by JoshuaKGoldberg** [comment]
Taking another look, I'm pretty sure this is actually a breaking change 🤔. Some users might be relying on the gap in rule coverage to import from restricted paths with `import = require()`.
